cli/src/main.rs

```diff
@@ -275,6 +289,47 @@ mod tests {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum GenerateProofError {
+    #[error("Unable to sign: {0}")]
+    Sign(#[from] didkit_cli::ssh_agent::SignError),
+    #[error("SSI: {0}")]
+    SSI(#[from] ssi::error::Error),
+    #[error("IO: {0}")]
+    IO(#[from] std::io::Error),
+}
+
+async fn generate_proof(
```

Comment:
Yes, `didkit_cli::ssh_agent` should probably be moved into `didkit` at least, so it could be used in `didkit_http`.

It could be moved into `ssi`. It depends on `tokio` which is already optionally depended on in `ssi`. I put it in didkit because it felt like an application integration (e.g. accessing the filesystem with UNIX sockets) - but it could also be considered as a (semi-standard) protocol.

To expose this functionality in the parts of DIDKit, some API changes will be needed, since currently it relies on CLI options and/or environmental variables.

An interface in ssi for ssh-agent-like things (to list keys, and request signatures) could allow key selection to stay in `ssi`.

Reply:
Okie dokie :+1:

Reply:
`ssh_agent` moved into `lib/` in 2cda8f4